src/Header/Authenticator.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef GROUP4_HEADER_AUTHENTICATOR_H_
+#define GROUP4_HEADER_AUTHENTICATOR_H_
+
+#include <string>
+
+//used to check if request is send from valid user
+//when getting request, requesthandler call authenticator::CheckAccount(account, password) check if it's valid.
+//if return true then keep handle request, false will deny the request.
+class Authenticator
+{
+public:
+    //singleton code convention method.
+    unique_ptr<Authenticator> GetInstance();
```

Comment:
If you are using singleton, then you cannot return a unique_ptr, because there can only be one instance of unique_ptr. Return a raw pointer instead.

Reply:
Done